tracing-core/src/dispatch.rs

```diff
@@ -577,6 +588,20 @@ impl Dispatch {
         me
     }
 
+    /// Creates a [`WeakDispatch`] from this `Dispatch`.
```

Comment:
maybe the docs for this function should also mention why you would want this (it could just be a link to the "Avoiding Memory Leaks" section in the docs for `Collect`)? it might also be nice to reference `Arc::downgrade` here...